src/components/CodeExample.js

```diff
@@ -130,28 +125,55 @@ const CodeSnippet = ({ codeSnippets, title, href }) => {
   );
   const [activeLang, setActiveLang] = React.useState("");
   const [isCopied, setCopy] = React.useState(false);
+  const [isHovered, setHover] = React.useState(false);
+  const [dimension, setDimension] = React.useState({ right: 0, top: 0 });
+
   const codeSnippetsArr = React.Children.toArray(codeSnippets);
   const selectedSnippet =
     codeSnippetsArr.find(
       (snippet) => snippet.props["data-language"] === activeLang,
     ) || codeSnippetsArr[0];
   const SelectedSnippetStr = mdxJsxToString(selectedSnippet);
 
+  const CopyToClipboardRef = React.useCallback((node) => {
```

Comment:
As React recommended, using `useCallback` to get the copy to clipboard icon's DOM position